### PR TITLE
add compatibility of functions

### PR DESCRIPTION
**Pull request by zespinosa97**

### CODE REVIEW

File: src/integration/java/com/smartcar/sdk/AuthTest.java [hunk 1]
```diff
@@ -74,9 +74,10 @@ public void testIsCompatible() throws Exception {
 
       String teslaVin = "5YJXCDE22HF068739";
       String royceVin = "SCA665C59HUX86700";
+      String scopes = "read_location read_odometer"; 
 
-      boolean teslaComp = this.authClient.isCompatible(teslaVin);
-      boolean royceComp = this.authClient.isCompatible(royceVin);
+      boolean teslaComp = this.authClient.isCompatible(teslaVin, scopes);
+      boolean royceComp = this.authClient.isCompatible(royceVin, scopes);
```
Comment (jerzzhang) [inline src/integration/java/com/smartcar/sdk/AuthTest.java]: Everything should be called `scope` (https://smartcar.com/docs/api#request-authorization)

File: src/main/java/com/smartcar/sdk/AuthClient.java [hunk 2]
```diff
@@ -365,10 +365,11 @@ public Auth exchangeRefreshToken(String refreshToken) throws SmartcarException {
    *
    * @throws SmartcarException when the request is unsuccessful
    */
-  public boolean isCompatible(String vin) throws SmartcarException {
```
Comment (jerzzhang) [inline src/main/java/com/smartcar/sdk/AuthClient.java]: Add Javadoc to define `scope`

File: src/main/java/com/smartcar/sdk/AuthClient.java [hunk 3]
```diff
@@ -365,10 +365,11 @@ public Auth exchangeRefreshToken(String refreshToken) throws SmartcarException {
...
+  public boolean isCompatible(String vin, String scopes) throws SmartcarException {
```
Comment (jerzzhang) [inline src/main/java/com/smartcar/sdk/AuthClient.java]: We would need to overload this function.
```java
public boolean isCompatible(String vin)
public boolean iscompatible(String vin, String scope)
```

And then write tests for both.

Comment (KarthikBhaskara) [reply src/main/java/com/smartcar/sdk/AuthClient.java]: ^yep

File: src/main/java/com/smartcar/sdk/AuthClient.java [hunk 4]
```diff
@@ -369,6 +369,36 @@ public boolean isCompatible(String vin) throws SmartcarException {
...
+   * Overloads isCompatible function to make scope optional
+   *
+   * @param vin the vin of the vehicle
+   * @param scope the space seperated string of scopes
+   *
+   * @return true if the vehicle is compatible
+   *
+   * @throws SmartcarException when the request is unsuccessful
+   */
+  public boolean isCompatible(String vin, String scope) throws SmartcarException {
```
Comment (KarthikBhaskara) [inline src/main/java/com/smartcar/sdk/AuthClient.java]: Ugh, I should've mentioned this before. scope should be a list of strings. We will then concat it into a space separated string before adding it onto the query params.

File: src/main/java/com/smartcar/sdk/AuthClient.java [hunk 5]
```diff
@@ -369,6 +369,36 @@ public boolean isCompatible(String vin) throws SmartcarException {
     HttpUrl url = HttpUrl.parse(this.urlApi).newBuilder()
         .addPathSegment("compatibility")
         .addQueryParameter("vin", vin)
+        .addQueryParameter("scope", scope)
```
Comment (KarthikBhaskara) [inline src/main/java/com/smartcar/sdk/AuthClient.java]: Did you mean to delete this?

File: src/test/java/com/smartcar/sdk/AuthClientTest.java [hunk 6]
```diff
@@ -685,10 +685,14 @@ public void testIsCompatible() throws Exception {
...
+    // Execute without scope
+    boolean compatibility = testSubject.isCompatible("vin");
+    // Execute with scope
+    boolean compatibilityWScopes = testSubject.isCompatible("vin", "scope");
+
 
     // Assertions
-    assertEquals(actual, expected.getCompatible());
+    assertEquals(compatibility, expected.getCompatible());
+    assertEquals(compatibilityWScopes, expected.getCompatible());
```
Comment (KarthikBhaskara) [inline src/test/java/com/smartcar/sdk/AuthClientTest.java]: What's happening here? :P

File: src/integration/java/com/smartcar/sdk/AuthTest.java [hunk 7]
```diff
@@ -82,6 +82,25 @@ public void testIsCompatible() throws Exception {
...
+    /**
+     * Test that the compatibility is returned given scope
+     *
+     * @throws Exception if compatibility cannot be obtained
+     */
+    @Test
+    public void testIsCompatibleScopes() throws Exception {
+
+      String teslaVin = "5YJXCDE22HF068739";
+      String royceVin = "SCA665C59HUX86700";
```
Comment (KarthikBhaskara) [inline src/integration/java/com/smartcar/sdk/AuthTest.java]: can you test with `WBAFR7C51DC821948`

File: src/main/java/com/smartcar/sdk/AuthClient.java [hunk 8]
```diff
@@ -357,19 +357,22 @@ public Auth exchangeRefreshToken(String refreshToken) throws SmartcarException {
   }
 
   /**
-   * Determine if a vehicle is compatible with Smartcar.
+   * Determine if a vehicle is compatible with Smartcar given scopes.
+   * Overloads isCompatible function to make scope optional
```
Comment (KarthikBhaskara) [inline src/main/java/com/smartcar/sdk/AuthClient.java]: update docs

File: src/integration/java/com/smartcar/sdk/AuthTest.java [hunk 9]
```diff
@@ -65,21 +65,22 @@ public void testRefreshToken() throws Exception {
...
+      String[] scope = new String[]{"read_location", "read_odometer"}; 
 
-      boolean teslaComp = this.authClient.isCompatible(teslaVin);
-      boolean royceComp = this.authClient.isCompatible(royceVin);
+      boolean teslaComp = this.authClient.isCompatible(teslaVin, scope);
+      boolean royceComp = this.authClient.isCompatible(royceVin, scope);
 
-      assertTrue(teslaComp);
-      assertFalse(royceComp);
+    //   assertTrue(teslaComp);
```
[the remark this answers is not in the document]
Comment (zespinosa97) [reply src/integration/java/com/smartcar/sdk/AuthTest.java]: yeah exactly, it'll keep failing until we actually launch compatibility

File: docs/com/smartcar/sdk/AuthClient.html [hunk 10]
```diff
@@ -786,14 +786,23 @@ <h4>exchangeRefreshToken</h4>
 <ul class="blockList">
 <li class="blockList">
 <h4>isCompatible</h4>
+<<<<<<< HEAD
```
Comment (KarthikBhaskara) [inline docs/com/smartcar/sdk/AuthClient.html]: merge conflict, my friend

File: src/integration/java/com/smartcar/sdk/AuthTest.java [hunk 11]
```diff
@@ -65,21 +65,22 @@ public void testRefreshToken() throws Exception {
...
 
     /**
-     * Test that the compatibility is returned.
+     * Test that the compatibility is returned given scope
      *
      * @throws Exception if compatibility cannot be obtained
      */
     @Test
-    public void testIsCompatible() throws Exception {
+    public void testIsCompatibleScopes() throws Exception {
```
Comment (KarthikBhaskara) [inline src/integration/java/com/smartcar/sdk/AuthTest.java]: just tesIsCompatible now

File: src/main/java/com/smartcar/sdk/AuthClient.java [hunk 12]
```diff
@@ -476,19 +476,22 @@ public Auth exchangeRefreshToken(String refreshToken) throws SmartcarException {
   }
 
   /**
-   * Determine if a vehicle is compatible with Smartcar.
+   * Determine if a vehicle is compatible with Smartcar given scopes.
+   * Vin and Scopes are required parameters
    *
    * @param vin the vin of the vehicle
+   * @param scope the space seperated string of scopes
```
Comment (KarthikBhaskara) [inline src/main/java/com/smartcar/sdk/AuthClient.java]: this is an array right?

File: docs/com/smartcar/sdk/AuthClient.html [hunk 13]
```diff
@@ -336,8 +336,8 @@ <h3>Method Summary</h3>
 </tr>
 <tr id="i15" class="rowColor">
 <td class="colFirst"><code>boolean</code></td>
-<td class="colLast"><code><span class="memberNameLink"><a href="../../../com/smartcar/sdk/AuthClient.html#isCompatible-java.lang.String-">isCompatible</a></span>(java.lang.String&nbsp;vin)</code>
-<div class="block">Determine if a vehicle is compatible with Smartcar.</div>
+<td class="colLast"><code><span class="memberNameLink"><a href="../../../com/smartcar/sdk/AuthClient.html#isCompatible-java.lang.String-">isCompatible</a></span>&#8203;(java.lang.String&nbsp;vin&nbsp;String[]&nbsp;scope)</code>
```
Comment (KarthikBhaskara) [inline docs/com/smartcar/sdk/AuthClient.html]: can you match these from https://github.com/smartcar/api-reference/pull/93, please

**Pull request merged**